### ISSUE TITLE
Add explicit dependency scala-reflect

### ISSUE THREAD

**Issue by olafurpg** [scalameta/scalafmt#1252]
Reported by @pjrt on Gitter against latest RC4
```
λ coursier bootstrap --standalone com.geirsson:scalafmt-cli_2.12:1.6.0-RC4 \
                    -r bintray:scalameta/maven \
                    -o ~/.local/bin/scalafmt-cli -f
λ scalafmt-cli --version
Exception in thread "main" java.lang.NoClassDefFoundError: scala/reflect/api/TypeCreator
    at fansi.ColorCategory.$anonfun$Full$1(Fansi.scala:851)
    at fansi.ColorCategory.$anonfun$Full$1$adapted(Fansi.scala:850)
    at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
    at scala.collection.immutable.Range.foreach(Range.scala:156)
    at scala.collection.TraversableLike.map(TraversableLike.scala:234)
    at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
    at scala.collection.AbstractTraversable.map(Traversable.scala:104)
    at fansi.ColorCategory.<init>(Fansi.scala:850)
    at fansi.Color$.<init>(Fansi.scala:717)
    at fansi.Color$.<clinit>(Fansi.scala)
    at fansi.Attr$.<init>(Fansi.scala:623)
    at fansi.Attr$.<clinit>(Fansi.scala)
    at fansi.Str$.<init>(Fansi.scala:373)
    at fansi.Str$.<clinit>(Fansi.scala)
    at org.scalafmt.config.OptIn$.$anonfun$surface$1(OptIn.scala:86)
    at pprint.TPrint$$anon$2.render(TPrint.scala:21)
    at org.scalafmt.config.OptIn$.surface$lzycompute(OptIn.scala:86)
    at org.scalafmt.config.OptIn$.surface(OptIn.scala:86)
    at org.scalafmt.config.OptIn.<init>(OptIn.scala:82)
    at org.scalafmt.config.ScalafmtConfig$.apply$default$4(ScalafmtConfig.scala:117)
    at org.scalafmt.config.ScalafmtConfig$.<init>(ScalafmtConfig.scala:192)
    at org.scalafmt.config.ScalafmtConfig$.<clinit>(ScalafmtConfig.scala)
    at org.scalafmt.cli.CliOptions$.apply$default$1(CliOptions.scala:77)
    at org.scalafmt.cli.CliOptions$.<init>(CliOptions.scala:14)
    at org.scalafmt.cli.CliOptions$.<clinit>(CliOptions.scala)
    at org.scalafmt.cli.Cli$.main(Cli.scala:66)
    at org.scalafmt.cli.Cli.main(Cli.scala)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at coursier.Bootstrap.main(Bootstrap.java:412)
Caused by: java.lang.ClassNotFoundException: scala.reflect.api.TypeCreator
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 32 more
```

Workaround for people using coursier is to add `scala-reflect` manually

```
coursier fetch com.geirsson:scalafmt-cli_2.12:1.6.0-RC4 org.scala-lang:scala-reflect:2.12.6
```

**Comment by pjrt** [comment]
Can confirm the workaround worked. 

```
λ coursier bootstrap --standalone com.geirsson:scalafmt-cli_2.12:1.6.0-RC4 org.scala-lang:scala-reflect:2.12.6 \
                    -r bintray:scalameta/maven \
                    -o ~/.local/bin/scalafmt-cli -f
λ scalafmt-cli --version
scalafmt 1.6.0-RC4
```

**Comment by olafurpg** [comment]
Fixed in #1284
